pkgs/development/python-modules/magic-wormhole/default.nix

```diff
@@ -15,49 +12,47 @@
 , tqdm
 , click
 , humanize
-, ipaddress
 , txtorcon
 , nettools
-, glibc
 , glibcLocales
 , mock
 , magic-wormhole-transit-relay
+, magic-wormhole-mailbox-server
 }:
 
 buildPythonPackage rec {
   pname = "magic-wormhole";
-  version = "0.10.5";
+  version = "0.11.2";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "9558ea1f3551e535deec3462cd5c8391cb32ebb12ecd8b40b36861dbee4917ee";
+    sha256 = "01fr4bi6kc6fz9n3c4qq892inrc3nf6p2djy65yvm7xkvdxncydf";
   };
 
-  checkInputs = [ mock magic-wormhole-transit-relay ];
-  buildInputs = [ nettools glibcLocales ];
-  propagatedBuildInputs = [ spake2 pynacl six attrs twisted autobahn automat hkdf tqdm click humanize ipaddress txtorcon ];
+  buildInputs = [ glibcLocales ];
+  propagatedBuildInputs = [ spake2 pynacl six attrs twisted autobahn automat hkdf tqdm click humanize txtorcon ];
+  checkInputs = [ mock magic-wormhole-transit-relay magic-wormhole-mailbox-server ];
 
   postPatch = ''
     sed -i -e "s|'ifconfig'|'${nettools}/bin/ifconfig'|" src/wormhole/ipaddrs.py
-    sed -i -e "s|if (os.path.dirname(os.path.abspath(wormhole))|if not os.path.abspath(wormhole).startswith('/nix/store') and (os.path.dirname(os.path.abspath(wormhole))|" src/wormhole/test/test_cli.py
-  '' + lib.optionalString (stdenv.isLinux) ''
-    # magic-wormhole will attempt to find all available locales by running
-    # 'locale -a'.  If we're building on Linux, then this may result in us
-    # running the system's locale binary instead of the one from Nix, so let's
-    # ensure we patch this.
-    sed -i -e 's|getProcessOutputAndValue("locale"|getProcessOutputAndValue("${glibc}/bin/locale"|' src/wormhole/test/test_cli.py
-  '' + lib.optionalString (pythonAtLeast "3.3") ''
-    sed -i -e 's|"ipaddress",||' setup.py
   '';
 
-  checkPhase = ''
-    export PATH="$PATH:$out/bin"
+  preCheck = ''
+    export PATH=$out/bin:$PATH
     export LANG="en_US.UTF-8"
     export LC_ALL="en_US.UTF-8"
-    ${python.interpreter} -m wormhole.test.run_trial wormhole
+    substituteInPlace src/wormhole/test/test_cli.py \
```

Comment:
This test had to have been fine if we're correcting it.

Reply:
sorry I missed that, I've updated the PR with a patch fixing the tests. Please let me know what you think